sdk/program/src/lib.rs

```diff
@@ -471,7 +471,6 @@ extern crate self as solana_program;
 
 pub mod account_info;
 pub mod address_lookup_table;
-pub mod alt_bn128;
```

Comment:
Um, what about gentle deprecation?
Maybe an alias to the new crate?

Reply:
Oh right, good idea 😅

Reply:
Actually, it seems difficult to import the new crate in `solana-program` due to circular dependency. It seems best to just leave the original logic in `solana-program` and add deprecation warnings pointing to the new crate. Wdyt?

Reply:
When https://github.com/anza-xyz/agave/pull/1827 lands, the circular dependency can be avoided by just moving the required syscall definitions from solana-program to solana-bn254

Reply:
#1827 has landed so you can just move the `define_syscall!` invocation and re-export the syscall like in #1656